macroslib/tests/expectations/cpp_code_cache.cpp

```diff
@@ -0,0 +1,23 @@
+r#"
+#ifdef __cplusplus
+extern "C" {
+#endif
+void CRustVecu8_free(struct CRustVecu8 v);
+#ifdef __cplusplus
+} // extern "C" {
+#endif
+
+#ifdef __cplusplus
+
+#include "rust_vec_impl.hpp"
+
+namespace org_examples {
+using RustVecu8 = RustVec<CRustVecu8, CRustVecu8_free>;
+}
+
+#endif
+
+// Automatically generated by flapigen
```

Comment:
"expectations test framwork" do not work in such way.

expectation works per file basic, not per the all generated files,
so you have to modify expection framework to check that this is whole file,
or may be add some new type, with `foreign_code(module = "rust_vec_u8.h"`,
and use this as marker of end of output.

Reply:
`that this is whole file`: Unfortunatelly, the test passes with the buggy code, as it appears at the end of the file.

I also tried `foreign_code #"xxx"#` but it apears too far.
I did not know/think about this version of `foreign_code`.
Let me check.

Reply:
In `collect_code_in_dir`, I added 2 markers:
- Before collected file : `<<< example_file.ext >>>`.
- After collected file : `>>> example_file.ext <<<`.

These markers allow to identify the file, from top or from bottom (or both).

Then, in `cpp_code_cache.cpp`, I get a marker for the end of file, instead of playing with the beg of next file:
```
namespace org_examples {
using RustVecu8 = RustVec<CRustVecu8, CRustVecu8_free>;
}

#endif

>>> rust_vec_u8.h <<<
"#;
```